src/table/virtualized-table/hooks/use-header-and-totals-height.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect } from 'react';
+import { VariableSizeList } from 'react-window';
+import { Column, PageInfo, TotalsPosition } from '../../../types';
+import { TableContext, useContextSelector } from '../../context';
+import { COMMON_CELL_STYLING } from '../../styling-defaults';
+import { GeneratedStyling } from '../../types';
+import { fontSizeToRowHeight } from '../../utils/styling-utils';
+import { MIN_BODY_ROW_HEIGHT, PADDING_TOP_BOTTOM } from '../constants';
+import useDynamicRowHeight from './use-dynamic-row-height';
+
+interface UseHeaderAndTotalsHeightProps {
+  columns: Column[];
+  columnWidths: number[];
+  pageInfo: PageInfo;
+  totalsPosition: TotalsPosition;
+  headerRef: React.RefObject<VariableSizeList<any>>;
+  totalsRef: React.RefObject<VariableSizeList<any>>;
+}
+
+const getBodyRowHeight = ({ fontSize = COMMON_CELL_STYLING.fontSize }: GeneratedStyling) =>
+  Math.max(MIN_BODY_ROW_HEIGHT, fontSizeToRowHeight(fontSize));
+
+const useHeaderAndTotalsHeight = ({
+  columns,
+  columnWidths,
+  pageInfo,
+  headerRef,
+  totalsRef,
+  totalsPosition,
+}: UseHeaderAndTotalsHeightProps) => {
+  const { layout, styling } = useContextSelector(TableContext, (value) => value.baseProps);
```

Comment:
We should move this into the `useDynamicRowHeight` hook. Well maybe need to keep styling here too but that would skip us having to pass two props to the hook

Reply:
that we can do

Reply:
since we are using the useDynamicRowHeight twice, we need to pass different styles. So I'll only move layout